src/transforms/eigenanalysis.jl

```diff
@@ -144,35 +156,56 @@ function eigenmatrices(transform, Y)
     S⁻¹ = V * Λ * transpose(V)
   end
 
-  S, S⁻¹
+  S[:, 1:d], S⁻¹[1:d, :]
```

Comment:
I am assuming you double checked carefully that the 1:d in the S^-1 case appears in the rows instead of in the columns.